alacritty/src/daemon.rs

```diff
@@ -111,3 +114,31 @@ pub fn foreground_process_path(
 
     Ok(cwd)
 }
+
+#[cfg(target_os = "openbsd")]
+pub fn foreground_process_path(
+    master_fd: RawFd,
+    shell_pid: u32,
+) -> Result<PathBuf, Box<dyn Error>> {
+    let mut pid = unsafe { libc::tcgetpgrp(master_fd) };
+    if pid < 0 {
+        pid = shell_pid as pid_t;
+    }
+    let name = [libc::CTL_KERN, libc::KERN_PROC_CWD, pid];
+    let mut buf = [0u8; libc::PATH_MAX as usize];
+    let result = unsafe {
+        libc::sysctl(
+            name.as_ptr(),
+            name.len().try_into().unwrap(),
+            buf.as_mut_ptr() as *mut _,
+            &mut buf.len() as *mut _,
+            ptr::null_mut(),
+            0,
+        )
+    };
+    if result != 0 {
+        return Err(io::Error::last_os_error().into());
+    }
+    let foreground_path = unsafe { CStr::from_ptr(buf.as_ptr().cast()) }.to_str()?;
+    Ok(PathBuf::from(foreground_path))
```

Comment:
```suggestion
    if result != 0 {
        Err(io::Error::last_os_error().into())
    } else {
        let foreground_path = unsafe { CStr::from_ptr(buf.as_ptr().cast()) }.to_str()?;
        Ok(PathBuf::from(foreground_path))
    }
```

Should do it either this way or add a newline between the early return and the successful return.